modules/framework/galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/routes/RunQueryRoute.java

```diff
@@ -312,6 +315,23 @@ private String buildResponseBody(List<RasRunResult> runs, int pageNum, int pageS
         return gson.toJson(runsPage);
 	}
 
+	private String findMatchingRequestor(String requestor) throws ResultArchiveStoreException, InternalServletException {
+		String matchedRequestor = null;
+
+		// Get all requestors in the service
+		List<String> requestorsList = getRequestors();
```

Comment:
Assuming this `getRequestors()` could throw a ResultArchiveStoreException, maybe we could have a try-catch and throw an appropriate InternalServletException instead?